File: CPB/uData.cpp
//---------------------------------------------------------------------------


#pragma hdrstop

#include "uData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma classgroup "Vcl.Controls.TControl"
#pragma link "dxmdaset"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "dxCore"
#pragma link "dxSkinsForm"
#pragma link "dxBackend"

#if defined(_WIN64)
  #pragma link "dxBackend.ConnectionString.SQL.O"
#else
  #pragma link "dxBackend.ConnectionString.SQL.OBJ"
#endif

#pragma resource "*.dfm"
TDataModule1 *DataModule1;
//---------------------------------------------------------------------------
const wchar_t* const DataFileName = L"data.dat";


__fastcall TDataModule1::TDataModule1(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TDataModule1::DataModuleCreate(TObject *Sender)
{
    if (FileExists(DataFileName))
		mdLayouts->LoadFromBinaryFile(DataFileName);
}
//---------------------------------------------------------------------------

void __fastcall TDataModule1::DataModuleDestroy(TObject *Sender)
{
	if (mdLayouts->RecordCount > 0)
		mdLayouts->SaveToBinaryFile(DataFileName);
}
//---------------------------------------------------------------------------

File: Delphi/uData.pas
unit uData;

interface

uses
  System.SysUtils, System.Classes, Data.DB, dxmdaset, cxClasses,
  dxCore, cxLookAndFeels,
  dxSkinsForm, dxBackend, dxBackend.ConnectionString.SQL;

type
  TDataModule1 = class(TDataModule)
    mdLayouts: TdxMemData;
    dsLayouts: TDataSource;
    mdLayoutsLayout: TBlobField;
    mdLayoutsName: TWideStringField;
    dxBackendDataConnectionManager1: TdxBackendDataConnectionManager;
    ReportsNWindConnectionString: TdxBackendDatabaseSQLConnection;
    dxSkinController1: TdxSkinController;
    procedure DataModuleCreate(Sender: TObject);
    procedure DataModuleDestroy(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  DataModule1: TDataModule1;

implementation

{%CLASSGROUP 'Vcl.Controls.TControl'}

{$R *.dfm}
const
  DataFileName = 'data.dat';

procedure TDataModule1.DataModuleCreate(Sender: TObject);
begin
  if FileExists(DataFileName) then
    mdLayouts.LoadFromBinaryFile(DataFileName)
end;

procedure TDataModule1.DataModuleDestroy(Sender: TObject);
begin
  if mdLayouts.RecordCount > 0 then
    mdLayouts.SaveToBinaryFile(DataFileName)
end;

end.
